exporters/pelorus/__init__.py

```diff
@@ -62,7 +62,9 @@ def setup_logging():
     )
     handler = logging.StreamHandler()
     handler.setFormatter(formatter)
-    root_logger.addHandler(handler)
+    # committime : containerimage threading.Thread().start() adds handler
+    if root_logger.hasHandlers():
+        root_logger.addHandler(handler)
```

Comment:
I think this will not add our handler for all other exporters

It should be
```py
if root_logger.hasHandlers():
    root_logger.handlers = []
root_logger.addHandler(handler)
```
not? This way, it checks if it already has another handler there, if yes, removes all of them before adding ours

Reply:
right.... fixed

Reply:
I just checked now and `pytest` also adds handlers when it runs (that is why unit and integration test are breaking).

I was able to unbreak the test with this changes
```diff
diff --git a/exporters/committime/app.py b/exporters/committime/app.py
index c0e14d2..a7ae786 100755
--- a/exporters/committime/app.py
+++ b/exporters/committime/app.py
@@ -271,9 +271,9 @@ class GitCommittimeConfig:
         )  # should be unreachable
 
 
-def set_up() -> AbstractCommitCollector:
+def set_up(prod: bool = True) -> AbstractCommitCollector:
     # TODO refactor: all exporters have same structure
-    pelorus.setup_logging()
+    pelorus.setup_logging(prod=prod)
     provider_config = load_and_log(CommittimeTypeConfig)
 
     dyn_client = pelorus.utils.get_k8s_client()
diff --git a/exporters/failure/app.py b/exporters/failure/app.py
index b298280..456c371 100755
--- a/exporters/failure/app.py
+++ b/exporters/failure/app.py
@@ -52,9 +52,9 @@ class FailureCollectorConfig:
         return load_and_log(PROVIDER_TYPES[self.tracker_provider])
 
 
-def set_up() -> AbstractFailureCollector:
+def set_up(prod: bool = True) -> AbstractFailureCollector:
     # TODO refactor: all exporters have same structure
-    pelorus.setup_logging()
+    pelorus.setup_logging(prod=prod)
 
     config = load_and_log(FailureCollectorConfig)
     collector = config.create()
diff --git a/exporters/pelorus/__init__.py b/exporters/pelorus/__init__.py
index 553b9c3..bb64c16 100644
--- a/exporters/pelorus/__init__.py
+++ b/exporters/pelorus/__init__.py
@@ -50,7 +50,7 @@ def _print_version():
 
 
 # region: logging setup
-def setup_logging():
+def setup_logging(prod: bool = True):
     _print_version()
     loglevel = utils.get_env_var("LOG_LEVEL", DEFAULT_LOG_LEVEL).upper()
     numeric_level = getattr(logging, loglevel, None)
@@ -63,7 +63,7 @@ def setup_logging():
     handler = logging.StreamHandler()
     handler.setFormatter(formatter)
     # committime : containerimage threading.Thread().start() adds handler
-    if root_logger.hasHandlers():
+    if prod and root_logger.hasHandlers():
         root_logger.handlers = []
     root_logger.addHandler(handler)
     root_logger.setLevel(numeric_level)
diff --git a/exporters/tests/__init__.py b/exporters/tests/__init__.py
index c563498..17eda8c 100644
--- a/exporters/tests/__init__.py
+++ b/exporters/tests/__init__.py
@@ -47,7 +47,7 @@ class MockExporter:
                 mock_kube_client.return_value.resources.get.side_effect = (
                     self.mock_kube_client
                 )
-                collector = self.set_up()
+                collector = self.set_up(prod=False)
             return collector
         finally:
             for key in arguments:
diff --git a/exporters/tests/test_config.py b/exporters/tests/test_config.py
index a0c8e13..582ebea 100644
--- a/exporters/tests/test_config.py
+++ b/exporters/tests/test_config.py
@@ -87,7 +87,7 @@ def test_loading_from_other():
 
 
 def test_logging(caplog: pytest.LogCaptureFixture):
-    pelorus.setup_logging()
+    pelorus.setup_logging(prod=False)
 
     @define(kw_only=True)
     class Loggable:
```
But if you want, you can revert the change in this file and I open a PR just for fixing the duplicated logs